libass/ass_utils.h

```diff
@@ -160,32 +161,15 @@ static inline int double_to_d22(double x)
     return lrint(x * 0x400000);
 }
 
-#define FNV1_32A_INIT 0x811c9dc5U
-#define FNV1_32A_PRIME 16777619U
-
-static inline uint32_t fnv_32a_buf(void *buf, size_t len, uint32_t hval)
+static inline uint32_t fnv_32a_buf(const void *buf, size_t len, uint32_t hval)
 {
-    unsigned char *bp = (unsigned char *) buf;
-    size_t n = (len + 3) / 4;
-
-    switch (len % 4) {
-    case 0: do { hval ^= *bp++; hval *= FNV1_32A_PRIME; //-fallthrough
-    case 3:      hval ^= *bp++; hval *= FNV1_32A_PRIME; //-fallthrough
-    case 2:      hval ^= *bp++; hval *= FNV1_32A_PRIME; //-fallthrough
-    case 1:      hval ^= *bp++; hval *= FNV1_32A_PRIME;
-               } while (--n > 0);
-    }
-
-    return hval;
+    static const uint64_t _wyp[5] = {0xa0761d6478bd642full, 0xe7037ed1a0b428dbull, 0x8ebc6af09c88c6e3ull, 0x589965cc75374cc3ull, 0x1d8e4e27c47d124full};
```

Comment:
I got this from upstream wyhash; not sure if it's essential to it passing smhasher so I figured I'd just keep it as-is.

Reply:
Current wyhash-upstream defines this default key in the header, making this definition unneccessary. *(There's also a function to generate other keys, haven't tested other keys though)*

https://github.com/wangyi-fudan/wyhash/blob/41043dc14fd4995e5e343c21cc14b5b075a9de63/wyhash.h#L116

The version in this pr, doesn't have this yet. Any particular reason you didn't use the most recent wyhash.h?

Reply:
The parameter is named `secret` if I’m not mistaken and the paper (in the upstream repo) calls it a “secret”, and it’s a parameter rather than just an internal constant… but then the current header has a mysterious `wyhash64` function that uses that value as a constant and seemingly so does `wyrand`… I don’t even mind the dense code so much, but I sure wish this was better _documented._

Reply:
I don't see a reason why this parameter should need to be secret; wyhash is not a cryptographic hash so even if this value were truly secret it would be unsuitable for security relevant use-case.
And as for generating a unique "secret": albeit referring to an older version, [wyhash#53](https://github.com/wangyi-fudan/wyhash/issues/53) suggests, that even when using `make_secret()` it takes many tries to obtain a usable "secret". Imho using the default `_wyp`-"secret" seems like the best option.

Reply:
It's indeed better to use unique secret for each run (per ASS_Library or Cache?). But upstream secret generation code is cumbersome and slow, not useful in runtime context. Why even secret values should be prime numbers? Even if they should be primes, there are much better algorithms for that. Also there is not much sense in making secret generation function inline and polluting the header, it's not in a fast path anyway.